quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/ConfigArgsConfigSource.java

```diff
@@ -48,20 +49,29 @@
 
     private static final Logger log = Logger.getLogger(ConfigArgsConfigSource.class);
 
+    private final boolean alwaysParseArgs;
+
     ConfigArgsConfigSource() {
         // higher priority over default Quarkus config sources
         super(parseArgument(), "CliConfigSource", 500);
+        alwaysParseArgs = isTestLaunchMode();
```

Comment:
Should we have a dedicated property? `parse.args`?

Reply:
I do not think so, this is really tests related. If we could avoid having more sys props better I would say.

Reply:
if it's really just test related 👍